File: implementations/Arrays/dynamic_array.py
class Array:

    def __init__(self):
        self.capacity = 2
        self.length = 0
        self.arr = [0] * 2

    # Method
    # insert n at end
    def pushback(self, n):
        if self.length == self.capacity:
            self.resize()
        self.arr[self.length] = n
        self.length += 1
    
    # create new array and double capacity
    def resize(self):
        self.capacity = 2 * self.capacity
        newArr = [0] * self.capacity

        # copy over elements
        for i in range(self.length):
            newArr[i] = self.arr[i]
        self.arr = newArr

    # remove last element in array
    def popback(self):
        if self.length > 0:
            self.length -= 1

    # get value
    def get(self, i):
        if i < self.length:
            return self.arr[i]
    
    # Insert at n
    def insert(self, i, n):
        if self.length == self.capacity:
            self.resize()
        for index in range(self.length - 1, i - 1, -1):
            self.arr[index + 1] = self.arr[index]

        self.arr[i] = n



# Create object instance
a = Array()
print(a.arr)

File: JSRamp/dynamic_array.js
class DynamicArray {
    /**
     * @constructor
     * @param {number} capacity
     */
    constructor(capacity) {
        this.capacity = capacity
        this.arr = new Array(this.capacity).fill(0)
        this.size = 0
    }

    /**
     * @param {number} i
     * @returns {number}
     */
    get(i) {
        return this.arr[i]
    }

    /**
     * @param {number} i
     * @param {number} n
     * @returns {void}
     */
    set(i, n) {
        this.arr[i] = n
    }

    /**
     * @param {number} n
     * @returns {void}
     */
    pushback(n) {
        if(this.size === this.capacity) {
            this.resize()
        }

        this.arr[this.size] = n
        this.size ++

    }

    /**
     * @returns {number}
     */
    popback() {
        if (this.size > 0) {
            this.size --
        }
        let temp = this.arr[this.size]
        this.arr[this.size] = 0
        return temp
    }

    /**
     * @returns {void}
     */
    resize() {
        this.capacity *= 2
        const new_arr = new Array(this.capacity).fill(0)
        for (let i = 0; i < this.size; i ++) {
            new_arr[i] = this.arr[i]
        }
        this.arr = new_arr
    }

    /**
     * @returns {number}
     */
    getSize() {
        return this.size
    }

    /**
     * @returns {number}
     */
    getCapacity() {
        return this.capacity
    }
}
